File: frontend/src/microservices/flight/api/FlightSuggestionsPlanner.ts
// 本文件定义 FlightSuggestionsPlanner，负责 flight 模块的建议查询入口。

import type { SearchSuggestionListPlannerResponse } from '@/microservices/flight/objects/SearchSuggestionListPlannerResponse'
import type { FlightSuggestionsPlannerRequest } from '@/microservices/flight/objects/FlightSuggestionsPlannerRequest'
import { executeJsonApiRequest } from '@/microservices/common/api/ApiTransport'

export const flightSuggestionsPlanner = (
  payload: FlightSuggestionsPlannerRequest,
): Promise<SearchSuggestionListPlannerResponse> =>
  executeJsonApiRequest('/FlightSuggestionsPlanner', 'POST', payload)

File: backend/src/main/scala/microservices/flight/api/FlightSuggestionsPlanner.scala
// FlightSuggestionsPlanner 是航班模块的业务入口，负责请求校验、流程编排和结果返回。

package com.typesafe.travel.flight.api

import com.typesafe.travel.flight.objects.*
import com.typesafe.travel.flight.tables.FlightPlannerRow

import cats.effect.IO
import com.typesafe.travel.api.routes.ConnectionApiPlan
import com.typesafe.travel.flight.tables.FlightSuggestionsPlannerPlainSql

import java.sql.Connection

object FlightSuggestionsPlanner extends ConnectionApiPlan[FlightSuggestionsPlannerRequest, SearchSuggestionListPlannerResponse]:
  override val name: String = "FlightSuggestionsPlanner"

  override def plan(input: FlightSuggestionsPlannerRequest, connection: Connection): IO[SearchSuggestionListPlannerResponse] =
    FlightSuggestionsPlannerPlainSql.suggestFlights(connection, input).map { rows =>
      SearchSuggestionListPlannerResponse(rows.map(toSuggestionResponse))
    }

  private def toSuggestionResponse(row: FlightPlannerRow): SearchSuggestionPlannerResponse =
    SearchSuggestionPlannerResponse(
      resourceType = "flight",
      value = row.flightId,
      title = s"${row.airlineName} ${row.flightNumber}".trim,
      subtitle = s"${row.departureAirport} -> ${row.arrivalAirport}"
    )
